crates/fs/benches/hamt.rs

```diff
@@ -0,0 +1,164 @@
+
+            for i in 0..50 {
+                node = black_box(node.set(i.to_string(), i, store).await.unwrap());
+            }
+        })
+    });
+}
+
+fn node_load_get(c: &mut Criterion) {
+    let mut store = MemoryBlockStore::default();
+    let cid: Cid = async_std::task::block_on(async {
+        let mut node = Rc::new(<Node<_, _>>::default());
+        for i in 0..50 {
+            node = node.set(i.to_string(), i, &mut store).await.unwrap();
+        }
+
+        let encoded_hamt = dagcbor::async_encode(&Hamt::with_root(node), &mut store)
+            .await
+            .unwrap();
+        store.put_serializable(&encoded_hamt).await.unwrap()
+    });
+
+    c.bench_function("Node load and get", |b| {
+        b.to_async(AsyncStdExecutor).iter(|| async {
+            let encoded_hamt = store.get_deserializable::<Vec<u8>>(&cid).await.unwrap();
+            let hamt: Hamt<String, i32> = dagcbor::decode(encoded_hamt.as_ref()).unwrap();
+
+            for i in 0..50 {
+                assert!(hamt
+                    .root
+                    .get(&i.to_string(), &store)
+                    .await
+                    .unwrap()
+                    .is_some());
+            }
+        })
+    });
+}
+
+fn node_load_remove(c: &mut Criterion) {
+    let mut store = MemoryBlockStore::default();
+    let cid: Cid = async_std::task::block_on(async {
+        let mut node = Rc::new(<Node<_, _>>::default());
+        for i in 0..50 {
+            node = node.set(i.to_string(), i, &mut store).await.unwrap();
+        }
+
+        let encoded_hamt = dagcbor::async_encode(&Hamt::with_root(node), &mut store)
+            .await
+            .unwrap();
+
+        store.put_serializable(&encoded_hamt).await.unwrap()
+    });
+
+    c.bench_function("Node load and remove", |b| {
+        b.to_async(AsyncStdExecutor).iter(|| async {
+            let encoded_hamt = store.get_deserializable::<Vec<u8>>(&cid).await.unwrap();
+            let mut hamt: Hamt<String, i32> =
+                black_box(dagcbor::decode(encoded_hamt.as_ref()).unwrap());
+
+            for i in 0..50 {
+                let (root, value) = hamt.root.remove(&i.to_string(), &store).await.unwrap();
+                assert!(value.is_some());
+                hamt.root = root;
+            }
+        })
+    });
+}
+
+macro_rules! hamt {
+    ($store:expr, { $( ($key:expr, $value:expr) $(, ($key_rest:expr, $value_rest:expr) )* $(,)? )? }) => {{
+        let root = Rc::new(Node::default());
+        $(
+            let root = root.set($key, $value, $store).await.unwrap();
+            $(
+                let root = root.set($key_rest, $value_rest, $store).await.unwrap();
+            )*
+        )?
+        black_box(Hamt::with_root(root))
+    }};
+}
+
+fn hamt_load_decode(c: &mut Criterion) {
+    let mut store = MemoryBlockStore::default();
+    let cid: Cid = async_std::task::block_on(async {
+        let hamt: Hamt<String, _> = hamt!(&mut store, {
+            ("foo".into(), 1),
```

Comment:
any magic number of 13? should we decode hamts at different sizes? We could maybe use proptest to generate various numbers (did something in previous libs).

Reply:
No reason at all. I should probably remove this static version in favor of a loop.
What do you think?
Also have you used `proptest` with criterion before? How do you deal with non-determinism?

Reply:
so, used a deterministic rvg, with a seed across runs: https://github.com/capsule-rs/capsule/blob/410696acb2e033cabb287750b36810753b54a59e/core/src/testils/rvg.rs#L38. We also created our own criterion iters for allowing for generated batches: https://github.com/capsule-rs/capsule/blob/410696acb2e033cabb287750b36810753b54a59e/core/src/testils/criterion.rs#L98.

Reply:
@appcypher a loop is prob fine; I was wondering if we wanted to decode across a set of different hamt sizes?